futures-select-macro/src/lib.rs

```diff
@@ -246,58 +246,74 @@ pub fn select(input: TokenStream) -> TokenStream {
         #complete_branch
     };
 
-    let await_and_select = if let Some(default_expr) = parsed.default {
+    let await_select_fut = if parsed.default.is_some() {
```

Comment:
Nice! This is actually how `select!` used to work, but then when I reorganized to introduce randomization I dropped support for this. Thanks for adding it back.